python/tvm/script/tir/ty.py

```diff
@@ -44,6 +44,9 @@ def __init__(self, vtype):
         self.type = vtype
 
     def evaluate(self):
+        if isinstance(self.type, tvm.ir.Type):
```

Comment:
I thought this change only impact `GenericPtrType` instead of `ConcreteType` here. Is that the case?

Reply:
This check gets hit when calling `.evaluate()` on the return type of `GenericPtrType.__getitem__`.  The return type is a `ConcreteType` that has been passed a `tvm.ir.PointerType`, which should then be returned directly when the concrete type is evaluated, rather than the usual behavior of wrapping it in a `tvm.ir.PrimType`.

Reply:
After thinking on it, I think that it would be cleaner if the type conversion happens during the call to `__init__`, rather than being delayed until the `.evaluate()` call.  That way, the `self.type` object has a consistent type, regardless of whether it was initialized with a string to represent a primitive, or was initialized with a `tvm.ir.Type` to represent that type.